packages/jest-cli/src/TestRunner.js

```diff
@@ -376,12 +385,75 @@ class TestRunner {
       this.addReporter(new CoverageReporter());
     }
 
-    this.addReporter(new SummaryReporter());
     if (config.notify) {
       this.addReporter(new NotifyReporter(this._startRun));
     }
   }
 
+  /**
+   * _setupDefaultReporters
+   * Method for setting up the default reporters
+   */
+  _setupDefaultReporters() {
+    const config = this._config;
+
+    this.addReporter(
+      config.verbose
+        ? new VerboseReporter(config)
+        : new DefaultReporter()
+    );
+
+    this.addReporter(new SummaryReporter());
+  }
+
+  /**
+   * Add Custom reporters to Jest
+   * Custom reporters can be added to Jest using the reporters option in Jest
+   * Config. The format for adding a custom reporter is following
+   *
+   * "reporters": [
+   *    ["reporterPath/packageName", { option1: 'fasklj' }], 
+   *    // Format if we want to specify options
+   *    
+   *    "reporterName"
+   *    // Format if we don't want to specify any options
+   * ]
+   */
+  _addCustomReporters(reporters: any) {
+    let reporterPath, reporterConfig;
+
+    reporters.forEach(entry => {
+      if (Array.isArray(entry)) {
+        [reporterPath, reporterConfig] = entry;
+      } else {
+        if (typeof entry !== 'string') {
+          throw new Error(
+            'Unexpected Custom Reporter Entry\n' +
+            'Report Entry should be of type:\n' +
+            `\t\t ${chalk.bold(chalk.green('String / Array'))}\n` +
+            'Received:\n' +
+            `\t\t ${chalk.bold(chalk.red(getType(entry)))}`
+          )
+        }
+
+        reporterPath = entry;
+      }
+
+      try {
+        const reporter = require(reporterPath);
+        this.addReporter(new reporter(reporterConfig || {}));
+      } catch (error) {
+        throw new Error(
+          `Failed to set up reporter: \n` +
+          `Reporter Path:\n` +
+          `\t\t${prettyFormat(reporterPath)}` + 
```

Comment:
Right now I am using this module `pretty-format` to do the formatting. This is temporary and afterwards I will figure out how I can make more changes.